src/experimental/tools/mediaCapabilitiesProber/api/index.ts

```diff
@@ -0,0 +1,164 @@
+ */
+
+import log from "../log";
+import {
+  IDisplayConfiguration,
+  IDRMConfiguration,
+  IMediaConfiguration,
+} from "../types";
+import probeMediaConfiguration, { IBrowserAPIS } from "./probeMediaConfiguration";
+
+/**
+ * A set of API to probe media capabilites.
+ * Each API allow to evalute a specific feature (HDCP support, decoding infos, etc)
+ * and relies on different browser API to probe capabilites.
+ */
+const mediaCapabilitiesProber = {
+
+  /**
+   * Set logger level
+   * @param {string} level
+   */
+  set LogLevel(level: string) {
+    log.setLevel(level);
+  },
+
+  /**
+   * Get logger level
+   * @returns {string}
+   */
+  get LogLevel() : string {
+    return log.getLevel();
+  },
+
+  /**
+   * Get HDCP status. Evaluates if current equipement support given
+   * HDCP revision.
+   * @param {string}
+   * @returns {Promise}
+   */
+  getStatusForHDCP(hdcp: string) : Promise<string> {
+    if (!hdcp) {
+      return Promise.reject("MediaCapabilitiesProbers >>> Bad Arguments: " +
+        "No HDCP Policy specified.");
+    }
+    const config = {
+      hdcp,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupportedWithFeatures",
+      "getStatusForPolicy",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      switch (result) {
+        case 0:
+          return "NotSupported";
+        case 1:
+          return "Unknown";
+        case 2:
+          return "Supported";
+        default:
+          return "NotSupported";
+      }
+    });
+  },
+
+  /**
+   * Get decoding capabilities from a given video and/or audio
+   * configuration.
+   * @param {Object} mediaConfig
+   * @returns {Promise}
+   */
+  getDecodingCapabilities(
+    mediaConfig: IMediaConfiguration
+  ) : Promise<string> {
+    const config = {
+      type: mediaConfig.type,
+      video: mediaConfig.video,
+      audio: mediaConfig.audio,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupported",
+      "isTypeSupportedWithFeatures",
+      "decodingInfos",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      switch (result) {
+        case 0:
+          return "NotSupported";
+        case 1:
+          return "MaybeSupported";
+        case 2:
+          return "Supported";
+        default:
+          return "NotSupported";
+      }
+    });
+  },
+
+  /**
+   * Tells if browser support deciphering with given drm type and configuration.
+   * @param {string} type
+   * @param {Object} drmConfig
+   * @returns {Promise}
+   */
+  isDRMSupported(
+    type: string,
+    drmConfig: IDRMConfiguration
+  ) : Promise<boolean> {
+    const config = {
+      keySystem: {
+        type,
+        configuration: drmConfig,
+      },
+    };
+    const browserAPIS: IBrowserAPIS[] = ["requestMediaKeySystemAccess"];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      return result === 2 ? true : false;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=254&issues=AWRgiSyQMt9UOhrw6nkR&open=AWRgiSyQMt9UOhrw6nkR)

[ISSUE_KEY]: AWRgiSyQMt9UOhrw6nkR